src/inmanta/module.py

```diff
@@ -278,13 +353,62 @@ def __init__(self, path: str) -> None:
         :param path: root git directory
         """
         self._path = path
-        self._meta = {}  # type: JsonType
+        self._metadata = self._get_metadata_from_disk()
+
+    def _get_metadata_from_disk(self) -> T:
+        metadata_file_path = self.get_metadata_file_path()
+
+        if not os.path.exists(metadata_file_path):
+            raise InvalidModuleException(f"Metadata file {metadata_file_path} does not exist")
+
+        with open(metadata_file_path, "r", encoding="utf-8") as fd:
+            try:
+                metadata_obj = yaml.safe_load(fd)
```

Comment:
Should we wrap invalid yaml error as well?

Reply:
Added.